wazo_calld/plugins/meetings/bus_consume.py

```diff
@@ -0,0 +1,100 @@
+# Copyright 2019-2021 The Wazo Authors  (see the AUTHORS file)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+import logging
+
+from wazo_calld.plugin_helpers.confd import Meeting
+from wazo_calld.plugin_helpers.exceptions import NoSuchMeeting
+
+from .schemas import participant_schema
+
+logger = logging.getLogger(__name__)
+
+
+class MeetingsBusEventHandler:
+    def __init__(self, confd, notifier, service):
+        self._confd = confd
+        self._notifier = notifier
+        self._service = service
+
+    def subscribe(self, bus_consumer):
+        bus_consumer.on_ami_event('ConfbridgeJoin', self._notify_participant_joined)
+        bus_consumer.on_ami_event('ConfbridgeLeave', self._notify_participant_left)
+
+    def _notify_participant_joined(self, event):
+        meeting_uuid = self._extract_meeting_uuid(event['Conference'])
+        if not meeting_uuid:
+            return
+
+        try:
+            meeting = Meeting.from_uuid(meeting_uuid, self._confd)
+        except NoSuchMeeting:
+            logger.debug(
+                'Ignored participant joining meeting %s: no such meeting', meeting_uuid
+            )
+            return
+
+        logger.debug('Participant joined meeting %s', meeting_uuid)
+        raw_participant = {
+            'id': event['Uniqueid'],
+            'caller_id_name': event['CallerIDName'],
+            'caller_id_number': event['CallerIDNum'],
+            'muted': event['Muted'] == 'Yes',
+            'answered_time': 0,
+            'admin': event['Admin'] == 'Yes',
+            'language': event['Language'],
+            'call_id': event['Uniqueid'],
+            'user_uuid': event.get('ChanVariable', {}).get('XIVO_USERUUID'),
+        }
+
+        participant = participant_schema.load(raw_participant)
```

Comment:
This is out of scope but I would like to see what the schema would look like if we move the "pre-parsing" to the schema class and do `participant_schema.load(event)`

Reply:
I like the idea, but we will not do that now :)